### ISSUE TITLE
Use function doc strings of functions to create API documetation for the services.

### ISSUE THREAD

**Issue by RatanShreshtha** [rorodata/firefly#68]
I would like to suggest that there should be a way to create API documentation for the services exposed through firefly using the function doc string for example:

```
def square(n):
    """Function that returns square of a number.

    Args:
         n (int): The first parameter.

    Returns:
        Square of the number n
    """
     return n**2
```

And once it is exposed as a service then the documentation will be available at some URL something like `/docs/square`

**Comment by anandology** [comment]
Yes, we need it. In fact, we are half way there. The API docs are available
at the / url as json. Need to expose it at /docs and also provide a way to
try it right there.


On Nov 4, 2017 3:37 PM, "Ratan Kulshreshtha" <notifications@github.com>
wrote:

I would like to suggest that there should be a way to create API
documentation for the services exposed through firefly using the function
doc string for example:

def square(n):
    """Function that returns square of a number.

    Args:
         n (int): The first parameter.

    Returns:
        Square of the number n
    """
     return n**2

And once it is exposed as a service then the documentation will be
available at some URL something like /docs/square

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/rorodata/firefly/issues/68>, or mute the thread

**Comment by RatanShreshtha** [comment]
I would like to add API browser for trying out API right there.
